artifactory/commands/utils/checkrunner.go

```diff
@@ -0,0 +1,215 @@
+	checks []PreCheck
+	// The status of the preformed checks on the current run
+	status *RunStatus
+	// Manage all the displayed progress of the run
+	displayBar *RunnerProgressBar
+}
+
+// Manage all the displayed progress of the run
+type RunnerProgressBar struct {
+	manager        *progressbar.ProgressBarMng
+	currentDisplay *progressbar.TasksProgressBar
+}
+
+// RunStatus - The status of the preformed checks in a run
+type RunStatus struct {
+	failures     uint
+	successes    uint
+	currentCheck string
+	startTime    time.Time
+}
+
+// RunArguments - The arguments of the run that is passed to the checks
+type RunArguments struct {
+	Context       context.Context
+	ServerDetails *config.ServerDetails
+	Repos         []string
+	ProgressMng   *progressbar.ProgressBarMng
+}
+
+// Creates a new empty runner
+func NewPreChecksRunner() *PreCheckRunner {
+	runner := &PreCheckRunner{}
+	return runner
+}
+
+// Add a check to the runner
+func (pcr *PreCheckRunner) AddCheck(check PreCheck) {
+	if check == nil {
+		return
+	}
+	pcr.checks = append(pcr.checks, check)
+}
+
+// Initialize a progress bar for running pre-checks
+func (pcr *PreCheckRunner) initProgressBar(status *RunStatus) (runnerProgress *RunnerProgressBar, err error) {
+	mng, shouldDisplay, err := progressbar.NewBarsMng()
+	if !shouldDisplay || err != nil {
+		return
+	}
+	runnerProgress = &RunnerProgressBar{manager: mng}
+	// The current check that is running
+	runnerProgress.currentDisplay = mng.NewStringProgressBar("Running check: ", func() string {
+		return color.Green.Render(status.currentCheck)
+	})
+	return
+}
+
+// Get all the server repositories base on include/exclude patterns
+func getAllRepos(context context.Context, serverDetails *config.ServerDetails, includeReposPatterns []string, excludeReposPatterns []string) ([]string, error) {
+	if serverDetails == nil {
+		return []string{}, nil
+	}
+	// Init source storage info manager
+	serviceManager, err := utils.CreateServiceManagerWithContext(context, serverDetails, false, 0, retries, retriesWaitMilliSecs)
+	if err != nil {
+		return []string{}, err
+	}
+	localRepos, err := utils.GetFilteredRepositoriesByNameAndType(serviceManager, includeReposPatterns, excludeReposPatterns, utils.Local)
+	if err != nil {
+		return []string{}, err
+	}
+	federatedRepos, err := utils.GetFilteredRepositoriesByNameAndType(serviceManager, includeReposPatterns, excludeReposPatterns, utils.Federated)
+	if err != nil {
+		return []string{}, err
+	}
+	return append(localRepos, federatedRepos...), nil
+}
+
+// Run all the checks and display the process
+func (pcr *PreCheckRunner) Run(context context.Context, serverDetails *config.ServerDetails, includeReposPatterns []string, excludeReposPatterns []string) (err error) {
+	log.Info(coreutils.PrintTitle(fmt.Sprintf("Running %d checks.", len(pcr.checks))))
+	// Init Run
+	repos, err := getAllRepos(context, serverDetails, includeReposPatterns, excludeReposPatterns)
+	if err != nil {
+		return
+	}
```

Comment:
I think the repositories handling should be in the longpropertycheck and not in the checkrunner.